### ISSUE TITLE
sold repo out of date?

### ISSUE THREAD

**Issue by mrcnski** [bluewhalesystems/sold#43]
1. The paid `sold` repo has not had activity for 5 months, as opposed to the free `mold` being last updated today.
2. The [build instructions](https://github.com/rui314/mold#compile-mold) do not work for `sold`. There are no tags so `git checkout v2.4.0` does not work.
3. I tried to build on the `main` branch but I get an error: 

```
$ ../install-build-deps.sh
../install-build-deps.sh: line 3: /etc/os-release: No such file or directory
```

As a paid user it would be nice to get the latest version of the code. I tried to build `mold` instead, but it also errored.

It would be nice to provide a good experience for paying users.

**Comment by rui314** [comment]
Sorry for the trouble. We are considering relicensing the sold linker as free software rather than as a commercial product, given the recent developments with the official Apple linker. Since Apple's linker has gained speed, it's difficult to obtain a sufficient number of users to sustain this project.

**Comment by acshi** [comment]
I'm also facing this `../install-build-deps.sh: line 3: /etc/os-release: No such file or directory` issue.

Is there any intermediate solution if I want to use sold in any capacity right now?

**Comment by satvikpendem** [comment]
@acshi @mrcnski You only have to run `install-build-deps.sh` on Linux, not macOS, which is likely what you're using as macOS does not have a `/etc/os-release` file, that is a Linux concept.

**Comment by rui314** [comment]
That's correct. That script is for Linux. For macOS, you can built it with CMake and Xcode.

**Comment by mrcnski** [comment]
In that case, the docs should be updated:

> The build instruction for sold is the same as for mold except that you check out https://github.com/bluewhalesystems/sold.git instead of https://github.com/rui314/mold.git.

**Comment by BLucky-gh** [comment]
> Sorry for the trouble. We are considering relicensing the sold linker as free software rather than as a commercial product, given the recent developments with the official Apple linker. Since Apple's linker has gained speed, it's difficult to obtain a sufficient number of users to sustain this project.

since sold is just the paid version of mold (I assume a fork?), are you going to just merge the differences back into mold or will sold just be the "no longer paid" version of the "not paid at all" mold with the only difference I can find being darwin support? The logical thing would be to just merge but the wording of that comments seems to imply that they'll stay separate...

**Comment by moritzuehling** [comment]
> That's correct. That script is for Linux. For macOS, you can built it with CMake and Xcode.

Is there any guide on that? My plan is to use sold with rust, and I'm not at all familar with Xcode. 

I must admit that I was a bit disappointed, I did buy sold and then found out I didn't even know how to build it :/ 

> Since Apple's linker has gained speed, it's difficult to obtain a sufficient number of users to sustain this project.

Is sold still faster? If yes, do you have numbers?

**Comment by rui314** [comment]
@moritzuehling Sorry for the trouble. I can refund you if you want. Honestly, it is impossible for a small company like us to compete with Apple if Apple is serious about it.

**Comment by X547** [comment]
If Sold is relicensed under MIT, is it possible to merge it back to Mold?

**Comment by rui314** [comment]
I wouldn't have the bandwidth to maintain the macOS part of the mold linker, so it's unlikely to happen.

**Comment by BLucky-gh** [comment]
What about adding a disclaimer that the macOS part is unmaintained and is provided "as is"?

Eventually someone might even send a PR

**Comment by perseoGI** [comment]
> What about adding a disclaimer that the macOS part is unmaintained and is provided "as is"?
> 
> Eventually someone might even send a PR

Yes that would be nice. I'll have liked to have a note in mold README warning that mold does nos support mach-o.
